src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -4613,5 +4646,803 @@ class var
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        var (_, x) = (1, 2);
+        System.Console.Write(x);
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "2");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var discard1 = GetDiscardDesignations(tree).First();
+            Assert.Null(model.GetDeclaredSymbol(discard1));
+            var tuple1 = (DeclarationExpressionSyntax)discard1.Parent.Parent;
+            Assert.Equal("var (_, x)", tuple1.ToString());
+            //Assert.Equal("(System.Int32, System.Int32)", model.GetTypeInfo(tuple1).Type.ToTestDisplayString()); // PROTOTYPE(wildcards) fix null type
+        }
+
+        [Fact]
+        public void UnderscoreLocalInDeconstructAssignment()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        int x, _;
+        (_, x) = (1, 2);
+        System.Console.Write($""{_} {x}"");
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "1 2");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var discard = GetDiscardIdentifiers(tree).First();
+            Assert.Equal("(_, x)", discard.Parent.Parent.ToString());
+            var symbol = (LocalSymbol)model.GetSymbolInfo(discard).Symbol;
+            Assert.Equal("System.Int32", symbol.Type.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void DiscardInForeach()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        foreach (var (_, x) in new[] { (1, ""hello"") }) { System.Console.Write(""1 ""); }
+        foreach ((_, var x) in new[] { (1, ""hello"") }) { System.Console.Write(""2""); }
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "1 2");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var discard1 = GetDiscardDesignations(tree).First();
+            Assert.Null(model.GetDeclaredSymbol(discard1));
+            var declaration1 = (DeclarationExpressionSyntax)discard1.Parent.Parent;
+            Assert.Equal("var (_, x)", declaration1.ToString());
+            //Assert.Equal("", model.GetTypeInfo(declaration1).Type.ToTestDisplayString()); // PROTOTYPE(wildcards) fix
+
+            var discard3 = GetDiscardIdentifiers(tree).First();
```

Comment:
Yes. Added a printout to make that visible.

---
In reply to: [88551775](https://github.com/dotnet/roslyn/pull/15192#discussion_r88551775) [](ancestors = 88551775)